src/components/reservations/ReservationReleasePDF.tsx

```diff
@@ -0,0 +1,93 @@
+import React, { useState } from 'react';
+import { Button, IconDocument, IconDownload } from 'hds-react';
+import { useTranslation } from 'react-i18next';
+
+import { useGetCustomerByIdQuery } from '../../redux/services/api';
+import { useDownloadFile } from '../../utils/useDownloadFile';
+import { useFileDownloadApi } from '../../utils/useFileDownloadApi';
+import { toast } from '../common/toast/ToastManager';
+
+import styles from './ReservationReleasePDF.module.scss';
+
+const T_PATH = 'components.reservations.ReservationReleasePDF';
+
+interface Props {
+  reservationId: number;
+  customerId: number;
+  size?: 'small' | 'normal';
+  disabled?: boolean;
+}
+
+const ReservationReleasePDF = ({
+  reservationId,
+  customerId,
+  size = 'normal',
+  disabled = false,
+}: Props): JSX.Element => {
+  const { t: translate } = useTranslation();
+  const t = (label: string) => translate(`${T_PATH}.${label}`);
+  const [isLoadingRelease, setIsLoadingRelease] = useState<boolean>(false);
+
+  // Non-essential, no need to wait for loading or check errors
+  // Used as supplemental data to name the PDF file
+  const { data: customer } = useGetCustomerByIdQuery(customerId.toString());
+
+  const releaseApiUrl = `/apartment_reservations/${reservationId}/release_pdf/`;
+
+  const preReleaseDownloading = () => setIsLoadingRelease(true);
+  const postReleaseDownloading = () => setIsLoadingRelease(false);
+
+  const onReleaseLoadError = () => {
+    setIsLoadingRelease(false);
+    toast.show({ type: 'error' });
+  };
+
+  const getReleaseFileName = (): string => {
+    const reservation = customer?.apartment_reservations?.find((ar) => ar.id === reservationId);
+
+    const projectName = reservation?.project_housing_company.replace(/\s/g, '-').toLocaleLowerCase();
+    const apartmentNumber = reservation?.apartment_number.replace(/\s/g, '').toLocaleLowerCase();
+    const identifier = projectName && apartmentNumber ? JSON.stringify(projectName + '_' + apartmentNumber) : '';
+
+    // Example output: "luovutushintalaskelma_as-oy-project-x_a01_2022-01-01.pdf"
+    return `luovutushintalaskelma_${identifier}_${new Date().toJSON().slice(0, 10)}.pdf`;
```

Comment:
This actually returns now `luovutushintalaskelma__as-oy-project-x_a01__2022-01-01.pdf` with double underscores around the `identifier`